avocado/utils/ssh.py

```diff
@@ -214,3 +214,37 @@ def quit(self):
         :rtype: bool
         """
         return self._master_command('exit')
+
+    def scp_connection(self):
+        try:
+            SCP_CLIENT_BINARY = path_utils.find_command('scp')
+        except path_utils.CmdNotFoundError:
+            SCP_CLIENT_BINARY = None
+        cmd = self._dash_o_opts_to_str(self.DEFAULT_OPTIONS)
+        cmd = "%s %s" % (SCP_CLIENT_BINARY, cmd)
+        return cmd
+
+    def scp_cmd(self, source, destination, reverse=False, recursive=False):
+        """
+        copy file to and from remote through scp session
+        :param source: local file
+        :type: str
+        :param destination: remote file location
+        :type: str
+        :param reverse: wether to copy to or from remote.if set
+                        to False copy to remote. and it True copy
+                        from remote.
+        :type: bool
+        :param recursive: scp option for copy file. if set to True
+                          copy files inside directory recursively.
+        :type: boot
+        """
+        options = self.scp_connection()
```

Comment:
IIUC, this method is just creating the command. So maybe `scp_connection()` isn't a good name. Also, if you handling the `CmdNotFoundError` and setting the command to `None` on `scp_command()` method, you are letting the exception be raised on the `process.run()` what makes me think: why handle that there? 

So maybe we don't need the `scp_connection()` method here. My suggestion would be something, like this:

```python
def copy_files(self, source, destination, recursive=False):
     cmd = path_utils.find_command('scp')
     options = self._dash_o_opts_to_str(self.DEFAULT_OPTIONS)
     if recursive:
        options += ' -r'
     options += "{} {}".format(source, destination)
     result = process.run("{} {}".format(cmd, options), ignore_status=True)
     return result.exit_status == 0
```
(I didn't test this code, just an idea/example)

Or you could use the same example here and put `process.run()` inside a try.. except to handle the exception here.

What do you think?

Reply:
@beraldoleal ok, i will change like this.